app/services/SdesService.scala

```diff
@@ -85,12 +89,48 @@ class SdesService @Inject()(sdesConnector: SdesConnector,
           )
         )
 
-        sdesConnector.notifySdes(payload)
+        sdesConnector.notifySdes(payload).map {
+          case res@SdesNotificationSuccess =>
+            logger.info(s"[SdesService] SDES notification sent for ${details.reference}")
+            res
+          case res@SdesNotificationFailure(body, status) =>
+            logger.error(s"[SdesService] SDES notification failed with status: $status and body: $body")
+            res
+        }
       })
     }
   }
 
-  def sdesCallback(stuff: String)(implicit hc: HeaderCarrier) = ???
+  def processCallback(sdesCallback: SdesCallback)(implicit hc: HeaderCarrier): Future[Unit] = {
```

Comment:
Would it be better to rename this callBack of what it does e.g. `submitAttachmentNonRepudiation`?

Reply:
That's not really all it does as it's also logging information (and later will be auditing it once it's decided how)
It also doesn't necessarily send anything to NRS as there's a case where the original NRS call failed resulting in all sdes notify and sdes callbacks not having an ID which is required to call submitAttachmentNonRepudiation. This case is agreed to be non blocking so the callback should still support it.

Reply:
👍